src/App.js

```diff
@@ -14,11 +15,95 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    goods: [...goodsFromServer],
+    isListShown: false,
+  };
+
+  showList= () => {
+    this.setState(state => ({
+      isListShown: !state.isListShown,
+    }));
+  }
+
+  reverse = () => {
+    this.setState(state => ({
+      goods: [...state.goods].reverse(),
+    }));
+  }
+
+  sortAlphabetically = () => {
+    this.setState(state => ({
+      goods: [...state.goods].sort((a, b) => a.localeCompare(b)),
+    }));
+  }
+
+  sortByLength = () => {
+    this.setState(state => ({
+      goods: [...state.goods].sort((a, b) => a.length - b.length),
+    }));
+  }
+
+  reset = () => {
+    this.setState(state => ({
+      goods: [...goodsFromServer],
+    }));
+  }
+
+  render() {
+    const { isListShown, goods } = this.state;
+
+    return (
+      <div className="App">
+        <button
+          type="button"
+          onClick={this.showList}
+          style={(isListShown)
+            ? { visibility: 'hidden' }
+            : { visibility: 'visible' }
+          }
+        >
+          Start
+        </button>
+        {(!isListShown)
+          ? ''
+          : (
+            <>
+              <h1>
+                Goods:
+              </h1>
+              <GoodsList goods={goods} />
```

Comment:
```suggestion
              <GoodsList goods={goodsFromServer} />
```
Let's pass initial goods to the `GoodsList` component and make all sortings there